WordPress/gutenberg
# Borders: Add BorderBoxControl component

Related: 
- https://github.com/WordPress/gutenberg/issues/35602
- https://github.com/WordPress/gutenberg/pull/37769

## Description

This PR adds a `BorderBoxControl` component that provides a better experience when configuring individual borders for different sides.

- `BorderBoxControl`
    - This leverages the `BorderControl` to allow the configuration of a single flat border or individual borders for each side.
    - Can toggle between flat or split border views
    - Mixed values for different side borders are supported
    - Editing a single border property while others are mixed is supported e.g. you can set a consistent width from the linked or "flat" border view after having previously selected different colors and styles for each side. Each side border's width would be updated but their colors and styles would remain intact.

## How has this been tested?
Via unit tests;

1. Run`npm run test-unit packages/components/src/border-box-control/test/index.js` 
2. Run `npm run test-unit packages/components/src/border-box-control/test/utils.js`

Manually:

1. Checkout this PR branch
2. Run `npm run storybook:dev`
3. [Check out the `BorderBoxControl`](http://localhost:50240/?path=/story/components-experimental-borderboxcontrol--default)

## Screenshots

<img src="https://user-images.githubusercontent.com/60436221/151919488-4eeac561-e8c1-4a9f-b767-3cdcaac7a6fd.gif" width="500" />

## Types of changes
New feature.

## Known Issues

- Tooltips within the popover both for the close button and color palette swatches impact layout. Will be addressed in follow-up.

## Checklist:
